ed_sensor_integration/src/kinect/image_buffer.cpp

```diff
@@ -91,33 +97,50 @@ bool ImageBuffer::waitForRecentImage(const std::string& root_frame, rgbd::ImageC
 
 // ----------------------------------------------------------------------------------------------------
 
-bool ImageBuffer::nextImage(const std::string& root_frame, rgbd::ImageConstPtr& image, geo::Pose3D& sensor_pose)
+bool ImageBuffer::nextImage(rgbd::ImageConstPtr& image, geo::Pose3D& sensor_pose)
+{
+    std::lock_guard<std::mutex> lg(recent_image_mutex_);
+    if(!recent_image_.first)
+    {
+        ROS_DEBUG("[IMAGE_BUFFER] No new image");
+        return false;
+    }
+
+    image = recent_image_.first;
+    sensor_pose = recent_image_.second;
+
+    recent_image_.first.reset(); // Invalidate the most recent image
+
+    return true;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+bool ImageBuffer::getMostRecentImageTF()
```

Comment:
It is currently missing the buffer functionality this thing was made for in the first place. If images and tf dont come in at the same time images will be discarded.

Reply:
Fixed by update using forward_list